src/intTest/java/uk/nhs/digital/nhsconnect/lab/results/uat/common/AbstractArgumentsProvider.java

```diff
@@ -0,0 +1,68 @@
+package uk.nhs.digital.nhsconnect.lab.results.uat.common;
+
+import lombok.SneakyThrows;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.ArgumentsProvider;
+import org.springframework.core.io.FileSystemResource;
+import org.springframework.core.io.Resource;
+import org.springframework.core.io.support.PathMatchingResourcePatternResolver;
+import org.springframework.core.io.support.ResourcePatternResolver;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+abstract class AbstractArgumentsProvider implements ArgumentsProvider {
+    protected static final String FHIR_FILE_ENDING = ".fhir.json";
+    protected static final String EDIFACT_FILE_ENDING = ".edifact.dat";
+
+    public abstract String getFolder();
+
+    public abstract Stream<? extends Arguments> provideArguments(ExtensionContext context) throws Exception;
+
+    protected String readResource(List<Resource> resources, String fileEnding) {
```

Comment:
We're always passing the value of `EDIFACT_FILE_ENDING` here for `fileEnding`, can we not just reference this?

Reply:
I think that was a leftover from the NHAIS method, I've made that suggested change now.